### ISSUE TITLE
[dev branch] external blob fetch apparent regression (in UUID external / 0ead3e4)

### ISSUE THREAD

**Issue by ixcat** [datajoint/datajoint-python#581]
The following does not seem to 'do':
```    
dj_schemas = dj.create_virtual_module('dj_schemas', 'test_blob')
keys = dj_schemas.Blobby().fetch(dj.key, limit=10)  # ok
(dj_schemas.Blobby() & keys[0])  # ok
(dj_schemas.Blobby() & keys[0]).fetch1()  # fails
print(rec)  # notreached
```
yielding:
```
$ ./blobfetch-t.py 
Connecting chris@someserver:3306
Traceback (most recent call last):
  File "./blobfetch-t.py", line 24, in <module>
    test_vmod_way()
  File "./blobfetch-t.py", line 14, in test_vmod_way
    (dj_schemas.Blobby() & keys[0]).fetch1()
  File "/home/ubuntu/.local/lib/python3.6/site-packages/datajoint/fetch.py", line 188, in __call__
    for name in heading.names)
  File "/home/ubuntu/.local/lib/python3.6/site-packages/datajoint/fetch.py", line 188, in <genexpr>
    for name in heading.names)
  File "/home/ubuntu/.local/lib/python3.6/site-packages/datajoint/fetch.py", line 46, in _get
    data = connection.schemas[attr.database].external[attr.store].get(uuid.UUID(bytes=data))
  File "/usr/lib/python3.6/uuid.py", line 134, in __init__
    raise TypeError('one of the hex, bytes, bytes_le, fields, '
TypeError: one of the hex, bytes, bytes_le, fields, or int arguments must be given

'''
```
It looks like the `Fetch` class's dispatch through `_get` into `external.get` for the case of `data=None` has a regression as the call to `external.get` now contains a `UUID` constructor which will not allow initialization from `None` as an argument, whereas previously this value was passed down to `external.get` which knew how to handle it. Following patch seems to fix :

```
$ git diff datajoint/fetch.py
diff --git a/datajoint/fetch.py b/datajoint/fetch.py
index e5b3fc8..3c7031a 100644
--- a/datajoint/fetch.py
+++ b/datajoint/fetch.py
@@ -43,7 +43,7 @@ def _get(connection, attr, data, squeeze, download_path):
     :return: unpacked data
     """
     if attr.is_external:
-        data = connection.schemas[attr.database].external[attr.store].get(uuid.UUID(bytes=data))
+        data = connection.schemas[attr.database].external[attr.store].get(uuid.UUID(bytes=data) if data else None)
     return (uuid.UUID(bytes=data) if attr.uuid else
             blob.unpack(data, squeeze=squeeze) if attr.is_blob else
             attach.save(data, download_path) if attr.is_attachment else data)
```
(PR forthcoming; issue filed for completeness)

**Comment by ixcat** [comment]
Actually @dimitri-yatsenko I realized that the PR could be written saving the call to `external.get` as follows:

current / pr #581:
`data = connection.schemas[attr.database].external[attr.store].get(uuid.UUID(bytes=data) if data is not None else data)
`
alternative:
`data = connection.schemas[attr.database].external[attr.store].get(uuid.UUID(bytes=data)) if data is not None else data`

let me know if you'd like me to re-PR

**Comment by eywalker** [comment]
I actually would have preferred the second solution

**Comment by ixcat** [comment]
resolved in #584
